blog/news-and-updates/2024-01-10-update.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Developer weekly update January 10, 2024
+description: This week, we have an exciting new release of Motoko, some news regarding increasing canister smart contract memory, and a call for community input!
+tags: [Devs]
+image: /img/blog/dev-update-blog-jan-10.jpg
+---
+
+# Developer weekly update January 10, 2024
+
+![January 10 2024](../../static/img/blog/dev-update-blog-jan-10.jpg)
+
+Hello developers, and welcome to this week's developer weekly update! This week, we have an exciting new release of Motoko, some news regarding increasing canister smart contract memory, and a call for community input! Let's get started.
+
+## Motoko v0.10.4
+
+This week, Motoko `v0.10.4` has been released! The release notes for this version are short, but for good reason: this Motoko update officializes the new incremental garbage collector after a successful beta testing phase! That's right, the incremental GC has officially been flagged for release outside of beta, meaning it is now regarded as safe for use in production. Some important notes about using the new incremental GC:
+
+- Thoroughly test the upgrade: Be sure to take adequate measures to verify that the amount of stable data is supported when upgrading your canister. Utilize different data volumes and heap sizes to conservatively determine the amount of stable data supported by your program. 
+
+- Monitor the heap size: Be sure to monitor `Prim.rts_memory_size()` and `Prim.rts_heap_size()` of your application while in production.
```

Comment:
@rossberg noted that encouraging the use of `Prim` sets a bad precedent. I would check in with @luc-blaeser to see if there is an alternate (less contentious) path

Reply:
Of course, Andreas' point is valid. We should expose the runtime metrics through the base library. This is something for the backlog. However, currently, this information can only be obtained by the use of `Prim`. Maybe we could write more generally "Be sure to monitor memory size and heap size of your code in production" such that the suggestion is more long-term applicable, namely also when the base library has been extended to offer this information.

Reply:
Revised per @luc-blaeser 's input.